src/Microsoft.AspNetCore.Razor.Evolution/TagHelpers/TagHelperFeature.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.Evolution.Legacy;
+
+namespace Microsoft.AspNetCore.Razor.Evolution.TagHelpers
+{
+    internal class TagHelperFeature : IRazorEngineFeature
```

Comment:
You described why it's currently `internal, but why does it need to be `public`? I think ideally it would be, but it's not a blocker for now.

I would also not assume that this class will be tied to `ITHDR` when we're done.

Also the naming is fine.